axiom/nr_distributed_trace.c

```diff
@@ -378,7 +378,16 @@ nrtime_t nr_distributed_trace_inbound_get_timestamp_delta(
     return 0;
   }
 
-  return nr_time_duration(txn_start, dt->inbound.timestamp);
+  return nr_time_duration(dt->inbound.timestamp, txn_start);
```

Comment:
Hi @b-viguier :)
Overall looks good, and thanks again for your engagement (and patience).

What was the rationale for switching the parameter signature?

There are quite a few places expecting with [this](https://github.com/newrelic/newrelic-php-agent/pull/164/files#diff-3c0fd593a6a423b5ac09884a07389563329c49e50a9347f1968bfe43a02b0d3bL381) order that don't account for the proposed new signature.

```
axiom/nr_exclusive_time.c:4
axiom/nr_txn.h:2
axiom/nr_distributed_trace.c:1
axiom/util_time.h:1
axiom/nr_segment_external.c:2
axiom/nr_app.c:1
axiom/nr_segment.c:4
axiom/nr_segment_datastore.c:2
axiom/nr_txn.c:3
axiom/nr_agent.c:1
axiom/tests/test_segment.c:1
axiom/tests/test_distributed_trace.c:3
axiom/tests/test_segment_tree.c:2
axiom/tests/test_time.c:3
axiom/tests/test_txn.c:3

agent/php_explain.c:2
agent/php_execute.c:2
agent/php_explain_mysqli.c:2
agent/lib_predis.c:1
agent/php_api_internal.c:1
```

Reply:
`nr_time_duration` is fine and well used in most case. Parameters still have the same meaning: `nr_time_duration(start, stop)` will return `stop - start` if `start < stop` (else `0`).

But I'm pretty sure that for **this** call, parameters are in the wrong order. This is certainly because it seems right to use `txn_start` as the `start` parameter of `nr_time_duration`, but it cannot! Here, we are computing the transport duration. Transport finishes when transaction starts. Chronologically, `dt->inbound.timestamp` is **before** `txn_start`, because it is the time where the transport started (timestamp included in HTTP headers).

So, to compute the right duration, we need to compute `txn_start - dt->inbound.timestamp`, _i.e._ `nr_time_duration(dt->inbound.timestamp, txn_start)`. I didn't noticed any similar misuse in other calls.

@zsistla is this clearer now? 😄